### PR TITLE
Support %{..} in toolbox <category name="...">

### PR DESCRIPTION
**Pull request by AnmAtAnm**
### Proposed Changes

Allow `%{..}` references in toolbox `<category name="...">`.

Still requires #1599 to render the generated section `<label>`.

### Reason for Changes

Support toolbox localization and runtime locale changes.

### Test Coverage

Tested in `vertical_playground.html`.

### CODE REVIEW

File: msg/messages.js [hunk 1]
```diff
@@ -272,8 +272,9 @@ Blockly.Msg.CATEGORY_EVENTS = 'Events';
 Blockly.Msg.CATEGORY_CONTROL = 'Control';
 Blockly.Msg.CATEGORY_SENSING = 'Sensing';
 Blockly.Msg.CATEGORY_OPERATORS = 'Operators';
-Blockly.Msg.CATEGORY_VARIABLES = 'Variables';
+Blockly.Msg.CATEGORY_VARIABLES = 'Data';
```
Comment (chrisgarrity) [inline msg/messages.js]: This is the label that translators are currently translating, and in Scratch 3.0 it should be 'Variables', so we shouldn't change this value.

File: blocks_vertical/default_toolbox.js [hunk 2]
```diff
@@ -518,11 +518,11 @@ Blockly.Blocks.defaultToolbox = '<xml id="toolbox-categories" style="display: no
...
     '</block>'+
   '</category>'+
-  '<category name="Data" id="data" colour="#FF8C1A" secondaryColour="#DB6E00" custom="VARIABLE">' +
+  '<category name="%{BKY_CATEGORY_VARIABLES}" id="data" colour="#FF8C1A" secondaryColour="#DB6E00" custom="VARIABLE">' +
   '</category>' +
-  '<category name="More" id="more" colour="#FF6680" secondaryColour="#FF4D6A" custom="PROCEDURE">' +
+  '<category name="%{BKY_CATEGORY_MYBLOCKS}" id="more" colour="#FF6680" secondaryColour="#FF4D6A" custom="PROCEDURE">' +
   '</category>' +
-  '<category name="Extensions" id="extensions" colour="#FF6680" secondaryColour="#FF4D6A" '+
+  '<category name="%{BKY_CATEGORY_EXTENSIONS}" id="extensions" colour="#FF6680" secondaryColour="#FF4D6A" '+
```
Comment (chrisgarrity) [inline blocks_vertical/default_toolbox.js]: I think we should just leave Extensions untranslated for now. It's not in scratch-gui (extensions provide their own name), and I'm trying not to give translators anything that they can't see in the gui. It leads to many questions.

File: core/toolbox.js [hunk 3]
```diff
@@ -688,7 +688,8 @@ Blockly.Toolbox.Category.prototype.createDom = function() {
   this.item_ = goog.dom.createDom('div',
       {'class': 'scratchCategoryMenuItem'});
   this.label_ = goog.dom.createDom('div',
-      {'class': 'scratchCategoryMenuItemLabel'}, this.name_);
+      {'class': 'scratchCategoryMenuItemLabel'},
+      Blockly.utils.replaceMessageReferences(this.name_));
```
Comment (chrisgarrity) [inline core/toolbox.js:692]: Looks good!

File: msg/messages.js [hunk 4]
```diff
@@ -272,8 +272,9 @@ Blockly.Msg.CATEGORY_EVENTS = 'Events';
...
 Blockly.Msg.CATEGORY_MYBLOCKS = 'My Blocks';
+Blockly.Msg.CATEGORY_EXTENSIONS = 'Extensions';
```
Comment (chrisgarrity) [inline msg/messages.js]: Remove this line - leaving 'extensions' untranslated above.

**Review by chrisgarrity — CHANGES_REQUESTED** [review]
It all makes sense. There's just a couple of changes needed to reduce the work (and potential confusion) for translators.

File: blocks_vertical/default_toolbox.js (hunk 2 above)
Comment (AnmAtAnm) [reply blocks_vertical/default_toolbox.js]: Understood. Done.

Do we need a comment to clarify for future devs?

File: msg/messages.js (hunk 1 above)
Comment (AnmAtAnm) [reply msg/messages.js]: Reverted here and in `scratch_mgs.js`.

File: msg/messages.js (hunk 4 above)
Comment (AnmAtAnm) [reply msg/messages.js]: Reverted here and in `scratch_mgs.js`.

**Review by AnmAtAnm — COMMENTED** [review]
Made requested changes.

**Review by chrisgarrity — APPROVED** [review]
Thanks @AnmAtAnm

**Pull request merged**